languages/scala/reference/README.md

```diff
@@ -0,0 +1,193 @@
+# Scala reference
+
+## Concepts
+
+The Scala concept exercises are based on concepts. The list below contains the concepts that have been identified for the Scala language.
+
+### Language-unique
```

Comment:
I don't think these are actually language-unique features. My Scala is terribly rusty. Are there features only Scala has implemented?

Reply:
I think so, too. Afaik `implicit`s are Scala-unique, but I would rather list the use cases here than the implicit keyword itself

Reply:
> Afaik implicits are Scala-unique, but I would rather list the use cases here than the implicit keyword itself

To me, it sounds like `implicits` is a Concept. Use cases are usually examples of how a Concept can be used, whereas this list should contain the actual Concept.

@leobenkel Could you remove the existing Concepts listed as being language-unique?

Reply:
What should I put here then ? 
Should I remove the content of this section and let other contributor fill it, for now ?

Reply:
@RafaelBa indicated that "implicits" is a Concept unique to Scala. I'd thus remove the existing entries and replace them with "implicits"

Reply:
I tried to come up with more stuff, but I'm frankly not sure if they really are unique to scala:  
- the `lazy` keyword - lets you mark a variable, so it will be evaluated on its first use
- the `trait` keyword - it can be used as interfaces, mixins, and partial implementation. Also all of it at the same time
- self types - gives you access to methods and fields of a class or interface, but will not extend the types signature
- path-dependent types - internal types of a base class differ from its counterpart in an inheriting class

From top to bottom they get more involved. I'm sorry if my explanations don't make sense, I tried to keep them short. If you want a deeper explanation or examples, I can provide that.

Anyway, what do you think? Are any of those Scala unique features that we want to add?

Reply:
I added them all :)